text/0191-swap-indexes-api.md

```diff
@@ -0,0 +1,126 @@
+
+##### 3.4.1.2. Response Definition
+
+When the request is in a successful state, Meilisearch returns the HTTP code `202 Accepted`. The response's content is the summarized representation of the received asynchronous task.
+
+```json
+{
+    "taskUid": 1,
+    "indexUid": null,
+    "status": "enqueued",
+    "type": "indexSwap",
+    "enqueuedAt": "2021-08-12T10:00:00.000000Z"
+}
+```
+
+> An `indexSwap` task is considered a global task; thus `indexUid` is null.
+
+See [Summarized `task` Object for `202 Accepted`](0060-tasks-api.md#summarized-task-object-for-202-accepted).
+
+##### 3.4.1.3. Errors
+
+###### 3.4.1.3.1. Synchronous Errors
+
+- 🔴 If the instance is secured by a master key, accessing this route without the `Authorization` header returns a [missing_authorization_header](0061-error-format-and-definitions.md#missing_authorization_header) error.
+- 🔴 If the instance is secured by a master key, accessing this route with a key that does not have permissions (missing `indexes.swap` action or having a value for the `indexes` field of a swap operation not being defined in the API Key `indexes` array) (i.e. other than the master key) returns an [invalid_api_key](0061-error-format-and-definitions.md#invalid_api_key) error.
+- 🔴 Omitting Content-Type header returns a [missing_content_type](0061-error-format-and-definitions.md#missing_content_type) error.
+- 🔴 Sending an empty Content-Type returns an [invalid_content_type](0061-error-format-and-definitions.md#invalid_content_type) error.
+- 🔴 Sending a different Content-Type than `application/json` returns an [invalid_content_type](0061-error-format-and-definitions.md#invalid_content_type) error.
+- 🔴 Sending an empty payload returns a [missing_payload](0061-error-format-and-definitions.md#missing_payload) error.
+- 🔴 Sending an invalid JSON payload returns a [malformed_payload](0061-error-format-and-definitions.md#malformed_payload) error.
+- 🔴 Sending an `indexes` array not containing **exactly** 2 indexUids for a swap operation object returns a [bad_request](0061-error-format-and-definitions.md#bad_request) error.
+
+###### 3.4.1.3.2. Asynchronous Errors
+
+- 🔴 Sending indexUids that do not exist within the `indexes` field of a swap operation returns an [index_not_found](0061-error-format-and-definitions.md#index_not_found) error.
+- 🔴 Sending an indexUid more than once in the request payload returns a [duplicate_index_found](0061-error-format-and-definitions.md#duplicate_index_found) error.
+
+## 4. Technical Details
+
+### 4.1. Swapping Data
+
+When indexes are swapped their data is exchanged. It concerns:
+
+- The documents
+- The settings
+- The tasks history
+    - `indexUid` field of tasks having been created before the `indexSwap` task `created_at` are updated to keep a practicable history. That is, it's possible to know why an index is in a given point in time state even if it has been swapped because it keeps the task history.
```

Comment:
Maybe a newline is needed or something but I don't get the sentence

Reply:
It would be good to explicitly say how they are updated. The implemented behaviour is that every mention of `indexA` in tasks that were enqueued before the index swap task is replaced by `indexB`, (and idem for `indexB` -> `indexA`).

Reply:
Would love a suggestion to replace that part!

Reply:
I let @loiclec make a suggestion because I don't even understand the original idea behind the sentence 😕

Reply:
Hi, 👋 
Just to be clear:
Once swapped, all mentions of `indexA` will be replaced by `indexB` **only for** `enqueued` tasks. All other tasks will show the "original" `indexA` in the `uid` field

Reply:
No it's the opposite. Let's say we have the following tasks:
```
1 succeeded createIndex A
2 succeeded createIndex B
3 enqueued swapIndex A B
4 enqueued addDocuments A
5 enqueued deleteIndex B
```
Then after task 3 (indexSwap) is processed, we have:
```
1 succeeded createIndex B
2 succeeded createIndex A
3 succeeded swapIndex A B
4 enqueued addDocuments A
5 enqueued deleteIndex B
```

Reply:
So the explanation could be something like:
```
An index swap between `index_a `and `index_b` will also replace every mention of `index_a` by `index_b` (and vice-versa) in the task history. Enqueued tasks are left unmodified.  
```

Reply:
Thanks, @loiclec!

Reply:
Thanks @loiclec! 🙏